src/python/pants/backend/python/lint/isort/subsystem.py

```diff
@@ -7,9 +7,10 @@
 
 class Isort(PythonToolBase):
   options_scope = 'isort'
-  default_version = 'isort==4.3.20'
+  default_version = 'isort>=4.3.21,<4.4'
   default_extra_requirements = ['setuptools']
   default_entry_point = 'isort.main'
+  default_interpreter_constraints = ["CPython>=3.6"]
```

Comment:
Technically, this is a lie. isort currently works when run with Python 2.7.

But, it doesn't matter what Python version you run isort with, unlike `flake8` and `bandit`. We can be confident that the user has Python 3.6+ on their machine because that's how they're running Pants. So, we can safely set the constraints to Python 3.6+.

(Also, the upcoming isort 5.0 will have this same constraint.)